security/cryptsetup-tests.py

```diff
@@ -34,33 +34,53 @@ def setUp(self):
         # Check for basic utilities
         smm = SoftwareManager()
         detected_distro = distro.detect()
-        deps = ['gcc', 'make']
-        if detected_distro.name in ['rhel', 'SuSE', 'fedora', 'centos',
-                                    'redhat']:
-            deps.extend(["autoconf", "automake", "gettext", "gettext-devel",
-                         "libtool", "device-mapper", "device-mapper-devel",
-                         "device-mapper-libs", "popt-devel", "json-c",
-                         "json-c-devel", "libblkid-devel", "libssh-devel"])
+        deps = ["gcc", "make", "autoconf", "automake", "gettext",
+                "gettext-devel", "libtool", "device-mapper", "popt-devel",
+                "device-mapper-devel", "libblkid-devel", "libssh-devel"]
+        if detected_distro.name in ['rhel', 'fedora', 'centos', 'redhat']:
+            deps.extend(["device-mapper-libs", "json-c", "json-c-devel"])
+        elif 'SuSE' in detected_distro.name:
+            deps.extend(["libjson-c-devel", "libjson-c3", "libuuid-devel"])
         else:
             self.cancel("Unsupported distro %s for cryptsetup package"
                         % detected_distro.name)
         for package in deps:
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://gitlab.com/cryptsetup/cryptsetup/"
-        git.get_repo(url, destination_dir=self.workdir)
-        os.chdir(self.workdir)
-        process.run("./autogen.sh")
-        process.run("./configure")
-        build.make(self.workdir)
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            default_url = "https://gitlab.com/cryptsetup/cryptsetup/"
```

Comment:
remove the default_url as we are hard_coding it in yaml

Reply:
@Naresh-ibm the idea here is to make sure the test runs with out yaml.. so we are good here